tools/mlir_bench/ov_model_gen.py

```diff
@@ -0,0 +1,247 @@
+class BaselineMLP(nn.Module):
+    def __init__(self, sizes_mnk, type=None):
+        super(BaselineMLP, self).__init__()
+        m = sizes_mnk[0]
+        n = sizes_mnk[1]
+        self.bias = torch.empty((m, n), dtype=type).data.fill_(0.01)
+        self.relu = nn.ReLU()
+    def forward(self, a, b):
+        c = torch.matmul(a, b)
+        c = torch.add(c, self.bias)
+        return self.relu(c)
+
+
+def baseline_MLP(model_desc: str, data_type: str, is_dynamic: bool) -> tuple[nn.Model, list]:
+    sizes = get_layer_sizes(model_desc)
+    assert len(sizes) == 3, "Invalid baseline MLP sizes"
+    mlp = BaselineMLP(sizes, get_torch_type(data_type))
+    input_shapes = get_layer_inputs(model_desc, is_dynamic)[0]
+    m = input_shapes[0]
+    n = input_shapes[1]
+    k = input_shapes[2]
+    ov_type = get_ov_type(data_type)
+    inputs = [(ov.PartialShape([m, k]), ov_type), (ov.PartialShape([k, n]), ov_type)]
+    return (mlp, inputs)
+
+
+def generate_baseline_model(model_desc: str, data_type: str, shape_type: str, file_name: str):
+    model_name = get_layer_name(model_desc)
+
+    if model_name == 'mlp':
+        baseline_tuple = baseline_MLP(model_desc, data_type, shape_type == 'dynamic')
+    else:
+        assert False, f"Unsupported baseline model data type {model_name}"
+
+    ov_model = ov.convert_model(baseline_tuple[0], input=baseline_tuple[1])
+    ov.save_model(ov_model, f"{file_name}")
+    return ov_model
+
+
+def main():
+    parser = argparse.ArgumentParser(
+                        prog='OV Model generator',
+                        description='Generate PyTorch model and export as OV .xml')
+    parser.add_argument('-l', '--layers', type=str.lower,
+                        help='Model layers description. For example:\
+                                -l="linear[64,32] relu[] linear[32,16] gelu[]"\
+                                -l="matmul[128,128,1024] add[128,128] relu[]"\
+                                -l="add[8,8] div[8,8]"')
+    parser.add_argument('-t', '--type', default='f32', type=str.lower,
+                        help='Data type: f32|f16|bf16|...')
+    parser.add_argument('-s', '--shapes', default='static', type=str.lower,
+                        help='Model shapes type: static|dynamic')
+    parser.add_argument('-n', '--name', default='temp.xml',
+                        help='Name for exported XML model')
+    parser.add_argument('-b', '--baseline', default=None, type=str.lower,
+                        help='Baseline pre-made model - overrides layers. For example:\
+                                -b=mlp[32,64,16]')
+    parser.add_argument('-p', '--print', action='store_true',
+                        help='Compile and print the model')
+    args = parser.parse_args()
+
+    if args.baseline is not None:
+        model = generate_baseline_model(args.baseline, args.type, args.shapes, args.name)
+    else:
+        model = generate_ov_model(args.layers, args.type, args.shapes, args.name)
+
+    if args.print:
```

Comment:
It only actually compiles if you print? How do we benchmark this?

Reply:
You provide saved model (.xml) to the benchmark_app that then does the compilation and execution.
Plus because we can use the c++ app, we avoid missing symbols from python.